lib/otel/context-manager.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+/**
+ * @see https://open-telemetry.github.io/opentelemetry-js/interfaces/_opentelemetry_api.ContextManager.html
+ */
+class ContextManager {
+  #ctxMgr
+
+  constructor(agent) {
+    this.#ctxMgr = agent.tracer._contextManager
+  }
+
+  active() {
+    return this.#ctxMgr.getContext()
+  }
+
+  bind(context, target) {
+    return boundContext.bind(this)
+
+    function boundContext(...args) {
+      return this.with(context, target, this, ...args)
+    }
+  }
+
+  /**
+   * Runs the callback within the provided context, optionally
+   * bound with a provided `this`.
+   *
+   * @param context
+   * @param callback
+   * @param thisRef
+   * @param args
+   */
+  with(context, callback, thisRef, ...args) {
+    return this.#ctxMgr.runInContext(context, callback, thisRef, args)
+  }
```

Comment:
Since this is really a pass through method, and we don't know if we will always have all of the parameters, I think it might be better written like so:

```suggestion
  with(...args) {
    return this.#ctxMgr.runInContext.apply(this.#ctxMgr, args)
  }
```

Reply:
this can't change. it actually fails with your suggest change. based on their API docs the first 3 args stay and then the rest are spread across the final arg: [see](https://open-telemetry.github.io/opentelemetry-js/interfaces/_opentelemetry_api.ContextManager.html#with)